File: 백준/Silver/1181. 단어 정렬/단어 정렬.py
import sys
input = sys.stdin.readline

arr = []
for i in range(int(input())):
    word=input().rstrip()
    arr.append(word)
arr=list(set(arr))
result = sorted(arr,key=lambda x: (len(x),x))

for i in result:
    print(i)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

input.shift();
let arr = [...new Set(input)];
arr = arr.sort((a, b) => a.length - b.length || a.localeCompare(b)).join("\n");
console.log(arr);
